dsc_lib/src/discovery/mod.rs

```diff
@@ -5,82 +5,99 @@ mod command_discovery;
 mod discovery_trait;
 
 use crate::discovery::discovery_trait::ResourceDiscovery;
-use crate::{dscresources::dscresource::DscResource, dscerror::DscError, dscerror::StreamMessageType};
-use regex::RegexBuilder;
+use crate::{dscresources::dscresource::DscResource, dscerror::DscError};
+use regex::{RegexBuilder, Regex};
+use std::collections::BTreeMap;
+use tracing::{debug, error};
 
 pub struct Discovery {
-    resources: Vec<DscResource>,
-    initialized: bool,
+    pub resources: BTreeMap<String, DscResource>,
 }
 
 impl Discovery {
     /// Create a new `Discovery` instance.
     ///
     /// # Errors
     ///
-    /// This function will return an error if the underlying discovery fails.
+    /// This function will return an error if the underlying instance creation fails.
+    ///
     pub fn new() -> Result<Self, DscError> {
         Ok(Self {
-            resources: Vec::new(),
-            initialized: false,
+            resources: BTreeMap::new(),
         })
     }
 
-    /// Initialize the discovery process.
+    /// List operation.
     ///
-    /// # Errors
+    /// # Panics
     ///
-    /// This function will return an error if the underlying discovery fails.
-    pub fn initialize(&mut self) -> Result<(), DscError> {
-        // TODO: this vec is leftover when we had multiple discovery types, we should
-        //       probably just have a single command discovery type
+    /// Will panic if `RegEx` creation fails.
```

Comment:
handle and fail gracefully instead of panic

Reply:
updated